### DISCUSSION TITLE
[knitr] RStudio: knitr support in RStudio | /knitr/demo/rstudio/

### DISCUSSION THREAD

**Discussion by yihui** [rbind/yihui#741] [Comments]
https://yihui.org/knitr/demo/rstudio/

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
Thanks for the feedback. Can you see a line `require(knitr); knit('knitr-minimal.Rnw')` in RStudio when you hit the compile button? Which version of knitr are you using? (see `packageVersion('knitr')` in R).

> Originally posted on 2012-02-27 19:49:02

**Reply by yihui** [reply]
Sounds weird. When you specify the weaving engine to be knitr, RStudio should load the knitr package and call knit(). I tried both the Ubuntu and Windows versions of the daily build, and they worked fine. Unfortunately I do not have a Mac to test with.

> Originally posted on 2012-02-28 00:40:31

**Reply by yihui** [reply]
Thanks for the solution. I will report this issue to RStudio developers.

> Originally posted on 2012-03-07 20:04:55

**Reply by yihui** [reply]
Hi, according to RStudio developers, this might help if you are inside an RStudio project: <http://www.rstudio.org/docs/authoring/latex_program> Also see the reply by @82c9d07618c90a1415c39a06550c728a:disqus below.

> Originally posted on 2012-03-08 22:10:47

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
Use knit() for Sweave(), and purl() for Stangle() after you library(knitr).

> Originally posted on 2012-06-15 21:43:50

**Reply by emelaktas** [reply]
I never got RStudio to knit pdf properly (using a Mac). I stick to Base R. Here is my knitloop (literally "knitloop.R") that I run on R to compile file.Rnw, which I edit in any editor (Textedit, R, Visual Studio Code). 

```
## knitr loop
setwd("~/Downloads/")
library("knitr")

file_name_part <- "2026_01_09"

  knit(paste0(file_name_part, ".Rnw"), output=paste0(file_name_part,"_me_v1.tex"))
  system(paste0("/Library/TeX/texbin/pdflatex ~/Downloads/", file_name_part, "_me_v1.tex"))

```

**Reply by yihui** [reply]
You can also call `knit2pdf()`:

```r
file_name_part <- "2026_01_09"
knitr::knit2pdf(paste0(file_name_part, ".Rnw"), output = paste0(file_name_part, "_me_v1.tex"))
```

**Reply by emelaktas** [reply]
Thank you so much for the suggestion. Unfortunately, I could not get it to work. I am not very good with environment variables, and I do not know how to fix "! /bin/sh: pdflatex: command not found". Thanks for trying to improve my code. I appreciate it.

**Reply by yihui** [reply]
You are welcome! It seems that `/Library/TeX/texbin` is not in your `PATH`. You can add it via:

```r
if (Sys.which('pdflatex') == '')
  Sys.setenv(PATH = paste(Sys.getenv('PATH'), '/Library/TeX/texbin'), sep = ':'))
```

This can be put in your `~/.Rprofile` so it will be automatically applied every time when R is started.

Alternatively, if you don't have a strong reason to use TeX Live (or MacTeX), you may try my TinyTeX (`tinytex::install_tinytex()`), which doesn't require you to modify `PATH`.

**Reply by yihui** [reply]
I just made a change in the R package {tinytex} so that you won't need to tweak the `PATH` variable by yourself: https://github.com/rstudio/tinytex/commit/e21398b383070fa98d72de5c24ffb1387eafa0b9 If you

```r
install.packages('tinytex', repos = c('https://rstudio.r-universe.dev', 'https://cloud.r-project.org'))
```

and restart R, you should be all set. No need to install TinyTeX if you don't want to.

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
File--Save with Encoding重新选UTF8保存一下再试试。Rmd可以用中文。多字节文档在Windows下一直是个麻烦，近期我会到Windows下再好好测试一下。建议你从Github重装一下formatR包：<https://github.com/yihui/formatR#readme>

> ——原帖发布于 2012-07-07 04:59:35

**Reply by yihui** [reply]
我刚历尽千辛万苦找出了一个在Windows下困扰我大半年的bug，现在在Windows殿下那里应该没有编码问题了（不管你用不用UTF8）。请按如下代码更新你的包再测试：

library(devtools)  
for (p in c('evaluate', 'formatR', 'knitr')) install_github(p, 'yihui')

在Windows下升级R没有别的办法，只能卸载重装（这不算太麻烦吧）。其实这完全可以通过程序自动实现，只是需要有能力、有精力且有兴趣的志愿者去写这个程序而已（我是三无人员）。

> ——原帖发布于 2012-07-08 01:23:55

**Reply by yihui** [reply]
不会吧……我在Win7下测试都没有问题啊。你确定上面三个包都安装成功了吗？

你可以装两个系统啊，我就是同时有Ubuntu和Windows，大部分时间里我都在Ubuntu下工作，实在需要用Windows就切过去用一下。

> ——原帖发布于 2012-07-09 00:49:31

**Reply by yihui** [reply]
是的，中文问题自从knitr 1.0.5就解决了，现在最新版本的RStudio也完美支持编码。

> ——原帖发布于 2013-03-03 20:31:01

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
What is an Rms file? You mean Rnw?

Regarding LaTeX, it depends on which operating system you use: for Windows, MikTeX; Mac OS X, MacTeX; Linux, TeXLive.

> Originally posted on 2014-10-12 05:22:20

[a comment by an automated account is not shown]
